MetaMask/metamask-extension
# fix: Fix types associated with auto lock timer call

## **Description**

We see the following error in sentry
![Screenshot from 2024-06-06 15-09-39](https://github.com/MetaMask/metamask-extension/assets/7499938/89d8e278-be26-46fb-9c01-493a9e42c41f)

The problem occurs when the user sets this field in advanced settings
![Screenshot from 2024-06-06 15-24-40](https://github.com/MetaMask/metamask-extension/assets/7499938/5c59576a-1c10-4712-8227-458ea4bdeba0)

If that field is set to empty, then this code in `/ui/pages/settings/advanced-tab/advanced-tab.component.test.js` would result in setting the background state property to the string `'0'`:
```
 if (autoLockTimeLimitBeforeNormalization === '') {
      this.setState({
        autoLockTimeLimitBeforeNormalization,
        autoLockTimeLimit: DEFAULT_AUTO_LOCK_TIME_LIMIT.toString(),
        lockTimeError: '',
      });
```

This `toString` call was introduced in https://github.com/MetaMask/metamask-extension/pull/19653, but it doesn't seem to be necessary. It also conflicts with the type of the prop that is passed in to initially set this component state property. Also, the `_resetTimer` function in the app-state controller clearly expects this value to be a number.

The fix is to not set this component state value as a string. There are a couple of other type related fixes that are not strictly necessary for fixing the bug, but improve the code.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25109?quickstart=1)

## **Manual testing steps**

1. Go to the page Advanced Settings
2. Set the "Auto-lock timer (minutes)" field to a non-zero integer (1 is best)
3. Wait that many minutes without interacting with MetaMask, you will automatically be locked out
4. Enter your password to unlock MetaMask
5. Go to Advanced Settings
6. Set  the "Auto-lock timer (minutes)" field to empty (delete the number in the field but don't enter anything else) and save it
7. Wait more than a minute, you won't be locked out.
8. Check the service worker console, there will not be an error.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
